Chocolatey.Cake.Recipe/Content/dependencyCheck.cake

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2023 Chocolatey Software, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+BuildParameters.Tasks.DependencyCheckTask = Task("Dependency-Check")
+    .WithCriteria(() => BuildParameters.ShouldRunDependencyCheck, "Skipping because DependencyCheck has been disabled")
+    .WithCriteria(() => !string.IsNullOrEmpty(BuildParameters.SonarQubeToken), "Skipping because SonarQube Token is undefined")
+    .IsDependentOn("Initialize-SonarQube")
+    .IsDependeeOf("Finalise-SonarQube")
+    .Does(() => RequireTool(ToolSettings.DependencyCheckTool, () =>
+{
+    DownloadFile(
+        "https://github.com/jeremylong/DependencyCheck/releases/download/v8.2.1/dependency-check-8.2.1-release.zip",
+        BuildParameters.RootDirectoryPath.CombineWithFilePath("dependency-check.zip")
+    );
```

Comment:
The downloading/extracting of this file is currently happening on each execution of this task.  I think we need to look at finding a way to perform this work only once, so that successive executions of the task doesn't re-download it when not required.

Perhaps we break the downloading/extracting of the tool into it's own task, leaving this task to _only_ do the execution of the tool?

Reply:
I think we should have a chat about this one.

Reply:
We chatted about this, and decided that this was fine as-is.  We will need to wait for further updates to this issue: https://github.com/burakince/DependencyCheck.Runner.Tool/issues/1 before deciding on how to proceed.